### PR TITLE
test(atomic): add TestRecsFixture & atomic-recs-interface tests

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-2076

### CODE REVIEW

File: packages/atomic/cypress/fixtures/test-recs-fixture.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+  initialize: (opts: {
+    accessToken: string;
+    organizationId: string;
+  }) => Promise<void>;
+  getRecommendations: () => void;
+  i18n: i18n;
+  language: string;
+};
+
+export class TestRecsFixture {
```
Comment (olamothe) [inline packages/atomic/cypress/fixtures/test-recs-fixture.ts:22]: It seems like quite a bit of copy paste from `TestFixture`... 

Could we refactor it a bit to share more code ? Should not be too complicated 🤔

File: packages/headless/src/features/configuration/search-configuration-actions-loader.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export interface SearchConfigurationActionCreators {
  * @returns An object holding the action creators.
  */
 export function loadSearchConfigurationActions(
-  engine: SearchEngine
+  engine: CoreEngine
```
Comment (olamothe) [inline packages/headless/src/features/configuration/search-configuration-actions-loader.ts:33]: Is this right ? What problem does this fix ?

File: packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx [hunk 3]
```diff
@@ -190,17 +191,36 @@ export class AtomicRecsInterface
...
+      return;
+    }
+
     this.commonInterfaceHelper.onAnalyticsChange();
   }
 
   public registerFieldsToInclude() {
     const fields = EcommerceDefaultFieldsToInclude.concat(
-      this.fieldsToInclude.split(',').map((field) => field.trim())
+      this.fieldsToInclude
```
Comment (olamothe) [inline packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx:220]: Should probably go with the `ArrayProp` utils here, for the fields to include property.

Comment (ThibodeauJF) [reply packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx:220]: Ah true!

File: packages/atomic/cypress/fixtures/test-recs-fixture.ts (hunk 1 above)
Comment (ThibodeauJF) [reply packages/atomic/cypress/fixtures/test-recs-fixture.ts:30]: Yes I've exported a bit more as of now

File: packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx (hunk 3 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx:220]: https://coveord.atlassian.net/browse/KIT-2092

File: packages/atomic/cypress/e2e/search-interface.cypress.ts [hunk 4]
```diff
@@ -113,24 +113,23 @@ describe('Search Interface Component', () => {
...
-        .withoutFirstAutomaticSearch()
+        .withLanguage('fr')
+        .withoutFirstIntercept()
         .init();
     });
 
-    it('should set locale for search request', async () => {
-      setLanguage('fr');
-      execSearch();
+    it('should set locale for search request', (done) => {
```
Comment (ThibodeauJF) [inline packages/atomic/cypress/e2e/search-interface.cypress.ts:121]: These tests were always true 😄

**Review by btaillon — APPROVED** [review]

File: packages/headless/src/features/configuration/search-configuration-actions-loader.ts (hunk 2 above)
Comment (btaillon) [reply packages/headless/src/features/configuration/search-configuration-actions-loader.ts:33]: If it only works for those two engines, then perhaps `SearchEngine | RecommendationEngine` would be better.

**Pull request merged**